internal/controller/apiextensions/composite/composition_ptf_test.go

```diff
@@ -1111,6 +1111,114 @@ func TestFunctionIODesired(t *testing.T) {
 	}
 }
 
+func TestWithKubernetesAuthentication(t *testing.T) {
+	errBoom := errors.New("boom")
+
+	type params struct {
+		c              client.Reader
+		namespace      string
+		serviceAccount string
+	}
+	type args struct {
+		ctx context.Context
+		fn  *v1.ContainerFunction
+		r   *fnv1alpha1.RunFunctionRequest
+	}
+	type want struct {
+		r   *fnv1alpha1.RunFunctionRequest
+		err error
+	}
+
+	cases := map[string]struct {
+		reason string
+		params params
+		args   args
+		want   want
+	}{
+		"GetServiceAccountError": {
+			reason: "We should return an error if we can't get the service account.",
+			params: params{
+				c: &test.MockClient{
+					MockGet: test.NewMockGetFn(errBoom),
+				},
+			},
+			want: want{
+				err: errors.Wrap(errBoom, errGetServiceAccount),
+			},
+		},
+		"GetSecretError": {
+			reason: "We should return an error if we can't get an image pull secret.",
+			params: params{
+				c: &test.MockClient{
+					MockGet: test.NewMockGetFn(nil, func(obj client.Object) error {
+						if _, ok := obj.(*corev1.Secret); ok {
+							return errBoom
+						}
+						return nil
+					},
+					),
+				},
+			},
+			args: args{
+				fn: &v1.ContainerFunction{
+					ImagePullSecrets: []corev1.LocalObjectReference{{Name: "cool-secret"}},
+				},
+			},
+			want: want{
+				err: errors.Wrap(errBoom, errGetImagePullSecret),
+			},
+		},
+		"Success": {
+			reason: "We should successfully parse OCI registry authentication credentials from an image pull secret.",
+			params: params{
+				c: &test.MockClient{
+					MockGet: test.NewMockGetFn(nil, func(obj client.Object) error {
+						if s, ok := obj.(*corev1.Secret); ok {
```

Comment:
we should check for secret name and return error if it is not equal to `cool-secret`. Otherwise, the test would pass even if `WithKubernetesAuthentication` has bug, i.e. it reads an arbitrary secret, independently what was passed argument.

Reply:
Good idea, will do.